trieste/models/keras/models.py

```diff
@@ -332,3 +332,129 @@ def optimize(self, dataset: Dataset) -> None:
 
         x, y = self.prepare_dataset(dataset)
         self.model.fit(x=x, y=y, **self.optimizer.fit_args)
+
+
+class MCDropout(KerasPredictor, TrainableProbabilisticModel):
+    def __init__(
+        self,
+        model: DropoutNetwork,
+        optimizer: Optional[KerasOptimizer] = None,
+        num_passes: int = 100,
+    ) -> None:
+
+        super().__init__(optimizer)
+
+        if not self.optimizer.fit_args:
+            self.optimizer.fit_args = {
+                "verbose": 0,
+                "epochs": 1000,
+                "batch_size": 32,
+                "callbacks": [
+                    tf.keras.callbacks.EarlyStopping(
+                        monitor="loss", patience=100, restore_best_weights=True
+                    ),
+                    tf.keras.callbacks.ReduceLROnPlateau(
+                        monitor="loss", factor=0.3, patience=20
+                    )
+                ],
+            }
+
+        if self.optimizer.loss is None:
+            self.optimizer.loss = "mse"
+
+        self._learning_rate = self.optimizer.optimizer.learning_rate.numpy()
```

Comment:
yes, this is a good solution, `DeepGaussianProcess` class in gpflux package does it as well...